connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -138,11 +149,13 @@ private void pause() {
 
                 case STARTED:
                     connector.stop();
+                    metrics.recordPaused();
                     // fall through
 
                 case INIT:
                     statusListener.onPause(connName);
                     this.state = State.STOPPED;
+                    metrics.recordPaused();
```

Comment:
There is a state change in the `catch` block that needs a `metrics.recordFailed()`

Reply:
I've redesigned this in #3911, but I'll move that into this PR. It changes the `ConnectorMetricsGroup` to implement and wrap the current `ConnectorStatus.Listener`. This way, the state changes for the supplied listener and the metrics group are handled the same way.